src/api/list_add.sql

```diff
@@ -0,0 +1,60 @@
+DROP FUNCTION IF EXISTS api.list_add(jsonb);
+
+CREATE OR REPLACE FUNCTION api.list_add(data jsonb)
+RETURNS void AS $$
+DECLARE  
+	class          jsonb;
+	obj_id 		   uuid;
+	value_to_add   jsonb; 
+	json_path      text[];
+	obj		 	   jsonb;
+	new_obj		   jsonb;
+	field_value    jsonb;
+	access_token   jsonb;
+
+	BEGIN
+		class := data->'class';
+		IF (class IS NULL) THEN
+			RAISE EXCEPTION 'The reclada object class is not specified';
+   		END IF;
+	
+   	
+		obj_id := (data->>'id')::uuid;
+		IF(obj_id IS NULL) THEN
+        	RAISE EXCEPTION 'The is no id';
+    	END IF;
+	
+		access_token := data->'access_token';
+		SELECT api.reclada_object_list(format(
+			'{"class": %s, "attrs": {}, "id": "%s", "access_token": %s}',
+        	class,
+        	obj_id,
+        	access_token
+    		)::jsonb) -> 0 INTO obj;
+    	
+    	IF (obj IS NULL) THEN
+        	RAISE EXCEPTION 'The is no object with such id';
+    	END IF;
+    	
+    	json_path := format('{attrs, %s}', data->>'field');
+    
+		value_to_add := data->'value';
+		IF (value_to_add IS NULL) THEN
+        	RAISE EXCEPTION 'There value should not be null';
+    	END IF;
+    	
+    	field_value := obj#>json_path;
+    	IF (field_value IS NULL) THEN
+        	SELECT jsonb_set(obj, json_path, value_to_add) || format('{"access_token": %s}', access_token)::jsonb
```

Comment:
The new value should be added as an item to an existing list; if there was no list in the original object, not even an empty list (NULL field_value), then either a new list with one item (value_to_add) should be created, or an exception should be raised, but definitely shouldn't a field of type `list' get scalar values, because it will not pass validation over the json schema afterwards. Either something like format('[%s]', value_to_add)::jsonb should be written here, or an exception should be raised.

Reply:
As we discussed value_to_add (new name: values_to_add) is array